### ISSUE TITLE
3-second delay upon signing into app

### ISSUE THREAD

**Issue by ltseng** [techmatters/terraso-mobile-client#435]
### App Version

1.0.40

### Platforms

- [ ] iOS
- [X] Android
- [ ] iOS simulator
- [ ] Android emulator

### Description

When I first tap Continue with Google and continue to select my account, the app returns me to the signin screen for about three seconds before proceeding to take me to the home screen. It looks as if I didn't sign in at all.

### Steps to Reproduce

1. Click on Continue with Google

### Expected behavior

After choosing an account, I should be redirected to the home screen map right away

### Actual behavior

I was taken back to the sign in page, waited three seconds, and then was taken to the home screen map

### Additional context

https://github.com/techmatters/terraso-mobile-client/assets/2730009/3ba3812e-3021-466a-bae0-2eb654c009b9

**Comment by shrouxm** [comment]
@ltseng this is presumably because the Terraso backend took a while to respond to the sign-in request. we could resolve this by either:
- navigating directly to the home screen before we hear from the Terraso backend. this could be confusing because if the request fails, we'll need to either then take you _back_ to the login screen, or leave you on the home screen but not signed in
- showing some kind of loading indicator on the login screen while waiting for the Terraso backend's response

i think we should wait to resolve this holistically when we make other changes we'll need to make to the login flow (e.g. anonymous users, different login providers, logging out, logging back into the app after logging out while offline, etc, etc, why does even login have so many complicated edge cases :sob:), since those might necessitate changing the login screen anyway

**Comment by ltseng** [comment]
Login was a significant part of my last job, so I understand it can be challenging to get right 😂

No worries though, I agree that this will be affected by other things and will affect other things in turn, which is why I made it a low-priority item to fix.

**Comment by ltseng** [comment]
Just noting that this is still happening as of version 48

**Comment by CourtneyLee333** [comment]
Here is a recording from today. It doesn't show my taps, but the first two login attempts were with my TM account. The third,  and only successful attempt, was with my personal gmail account. I've repeated this multiple times yesterday and today, waiting for much longer with each attempt than what I did in this recording.  

https://github.com/techmatters/terraso-mobile-client/assets/8915145/3d67af42-10cf-4281-adfa-6683fbfe7725
